sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionUtilityToolFilter.java

```diff
@@ -0,0 +1,196 @@
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import edu.umich.its.lti.utils.PropertiesUtilities;
+
+
+
+public class SectionUtilityToolFilter implements Filter {
+
+	private static Log M_log = LogFactory.getLog(SectionUtilityToolFilter.class);
+	
+	private static final String OU_GROUPS = "ou=Groups";
+	private static final String LDAP_CTX_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+	protected static final String SYSTEM_PROPERTY_FILE_PATH_SECURE = "sectionsToolPropsPathSecure";
+	protected static final String SYSTEM_PROPERTY_FILE_PATH_LESS_SECURE = "sectionsToolPropsPathLessSecure";
+	protected static final String PROPERTY_CANVAS_ADMIN = "canvas.admin.token";
+	protected static final String PROPERTY_CANVAS_URL = "canvas.url";
+	protected static final String PROPERTY_USE_TEST_URL = "use.test.url";
+	protected static final String PROPERTY_LDAP_SERVER_URL = "ldap.server.url";
+	private static final String PROPERTY_AUTH_GROUP = "mcomm.group";
+	private static final String TEST_USER = "testUser";
+	private String providerURL = null;
+	private String mcommunityGroup = null;
+	private boolean isTestUrlEnabled=false;
+	protected static Properties appExtSecurePropertiesFile=null;
+	protected static Properties appExtLessSecurePropertiesFile=null;
+	private static final String FALSE = "false";
+	
+	@Override
+	public void init(FilterConfig filterConfig) throws ServletException {
+		M_log.debug("Filter Init(): Called");
+		getExternalAppProperties();
+		
+	}
+
+	@Override
+	public void doFilter(ServletRequest request, ServletResponse response,
+			FilterChain chain) throws IOException, ServletException {
+		M_log.debug("doFilter: Called");
+		HttpServletRequest useRequest = (HttpServletRequest) request;
+		HttpServletResponse useResponse=(HttpServletResponse)response;
+		if(!checkForAuthorization(useRequest)) {
+			useResponse.sendError(403);
+			return;
+		}
+		chain.doFilter(useRequest, response);
+	}
+
+	@Override
+	public void destroy() {
+		M_log.debug("destroy: Called");
+	}
+	
+	protected void getExternalAppProperties() {
+		M_log.debug("getExternalAppProperties(): called");
+		String propertiesFilePathSecure = System.getProperty(SYSTEM_PROPERTY_FILE_PATH_SECURE);
+		String propertiesFilePathLessSecure = System.getProperty(SYSTEM_PROPERTY_FILE_PATH_LESS_SECURE);
+		if (!isEmpty(propertiesFilePathSecure)) {
+			appExtSecurePropertiesFile=PropertiesUtilities.getPropertiesObjectFromURL(propertiesFilePathSecure);
+			if(appExtSecurePropertiesFile!=null) {
+				isTestUrlEnabled = Boolean.parseBoolean(appExtSecurePropertiesFile.getProperty(SectionUtilityToolFilter.PROPERTY_USE_TEST_URL,FALSE));
+			}else {
+				M_log.warn("Failed to load secure application properties from sectionsToolPropsSecure.properties for SectionsTool");
+			}
+			
+		}else {
+			M_log.error("File path for (sectionsToolPropsPathSecure.properties) is not provided");
```

Comment:
Probably a moot point if you merge the 2 files, but why is this an error, while the lack of the file is a warning? They both result in the same condition, which I don't think is fatal, so likely just a warning here.